libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -234,12 +234,75 @@ void IRGeneratorForStatements::endVisit(Return const& _return)
 
 void IRGeneratorForStatements::endVisit(UnaryOperation const& _unaryOperation)
 {
-	if (type(_unaryOperation).category() == Type::Category::RationalNumber)
+	Type const& resultType = type(_unaryOperation);
+	Token const op = _unaryOperation.getOperator();
+
+	if (resultType.category() == Type::Category::RationalNumber)
+	{
 		defineExpression(_unaryOperation) <<
-			formatNumber(type(_unaryOperation).literalValue(nullptr)) <<
+			formatNumber(resultType.literalValue(nullptr)) <<
 			"\n";
+	}
+	else if (resultType.category() == Type::Category::Integer)
+	{
+		string modifiedValue = m_context.newYulVariable();
+
+		if (op == Token::Inc || op == Token::Dec)
+		{
+			auto const* intType = dynamic_cast<IntegerType const*>(&resultType);
+
+			solAssert(!!m_currentLValue, "LValue not retrieved.");
+			string fetchValueExpr = m_currentLValue->retrieveValue();
+			m_code << "let " <<
+				modifiedValue << " := " <<
+				(op == Token::Inc ?
+					m_utils.incrementFunction(*intType) :
+					m_utils.decrementFunction(*intType)) <<
+				"(" << fetchValueExpr << ")\n";
+
+			string returnValue;
+
+			if (_unaryOperation.isPrefixOperation())
```

Comment:
I think this still looks rather complicated if you first determine which value you will return, then store something and then only use the value you determined earlier.
What about:

```
let <originalValue> := <fetchExpr>
let <modifiedValue> := <incr/decr>(<originalValue>)
<storeValue>(<modifiedValue>)
```
`defineExpression(_unaryOperation) << (prefix ? originalValue : modifiedValue) << "\n";`

Reply:
Yeah, that looks much better indeed.